src/main/java/io/github/hapjava/impl/http/impl/BinaryHandler.java

```diff
@@ -27,9 +25,12 @@ public BinaryHandler(HomekitClientConnection connection) {
   @Override
   protected void encode(ChannelHandlerContext ctx, ByteBuf msg, ByteBuf out) throws Exception {
     if (started) {
+      logger.debug(
+          String.format(
+              "Sending data [%s]:%n%s",
+              ctx.channel().remoteAddress().toString(), msg.toString(StandardCharsets.UTF_8)));
```

Comment:
This is unnecessarily creating objects even when debug logging is off.

Reply:
Good point. Pushed an update.